scheduler/scheduler.go

```diff
@@ -526,29 +629,54 @@ func (driver *MesosSchedulerDriver) Start() (mesos.Status, error) {
 	driver.setStatus(mesos.Status_DRIVER_RUNNING)
 	driver.setStopped(false)
 
-	log.Infoln("Mesos scheduler driver started with PID=", driver.self.String())
+	log.Infof("Mesos scheduler driver started with PID=%v", driver.self)
 
+	// start the detector AFTER we have a self pid from the messenger, otherwise things get ugly.
```

Comment:
"start the detector..." I thought the detector would have already been started from the detector/zoo/plugin.init() no ? I ran into the same ugliness.